packages/strapi-plugin-upload/services/image-manipulation.js

```diff
@@ -69,7 +69,8 @@ const optimize = async buffer => {
   }
 
   const sharpInstance = autoOrientation ? sharp(buffer).rotate() : sharp(buffer);
-  return sharpInstance
+
+  const optimized = await sharpInstance
     .toBuffer({ resolveWithObject: true })
     .then(({ data, info }) => ({
       buffer: data,
```

Comment:
can't you just do buffer.length < data.length here ?  that would help us keep the simple return of promise

Reply:
Had to edit the `size` param below too, but indeed it looks better that way.